src/NakamaSync/Var.cs

```diff
@@ -263,6 +266,32 @@ private ValidationStatus TryHostIntercept(IUserPresence source, ISerializableVar
             return newStatus;
         }
 
+        private void SetValueFromIncoming(T incomingValue)
+        {
+            if (incomingValue == null)
+            {
+                Value = default(T);
+                return;
+            }
+            
+            var type = incomingValue.GetType();
+            if (Value != null && type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+            {
+                var incomingDictionary = (IDictionary) incomingValue;
+                var newDictionary = (IDictionary) Value;
+                foreach (var key in incomingDictionary.Keys)
+                {
+                    newDictionary[key] = incomingDictionary[key];
```

Comment:
We'll need to handle removal of dictionary values as well. Currently each client's local dictionary can only grow.